controlplane/kubeadm/internal/cluster.go

```diff
@@ -93,7 +97,20 @@ func (m *Management) GetWorkloadCluster(ctx context.Context, clusterKey client.O
 	}
 	restConfig.Timeout = 30 * time.Second
 
-	c, err := client.New(restConfig, client.Options{Scheme: scheme.Scheme})
+	clientCache, err := cache.New(restConfig, cache.Options{Scheme: scheme.Scheme})
+	if err != nil {
+		return nil, &RemoteClusterConnectionError{Name: clusterKey.String(), Err: err}
+	}
+
+	newClient := util.DelegatingClientFuncWithUncached(
```

Comment:
@sbueringer Sorry for the confusion, could we actually try to just use the plain `client.New` without the delegating client and retrieve with PartialObjectMetadata directly? It _should_ work considering that it has support for it https://github.com/kubernetes-sigs/controller-runtime/blob/release-0.5/pkg/client/client.go#L111

Reply:
Oh, yeah will try. At least I'm learning a lot :)

Reply:
Works, tested it locally with all relevant branches of IsKubernetesVersionSupported and it uses the metaclient as expected.